feat: ssh tunnel shortcuts

pgierz opened the pull request.

siligam commented on src/pymorize/ssh_tunnel.py:
```diff
@@ -0,0 +1,40 @@
+    on a remote compute node, routed through a gateway server.
+    """
+    ssh_command = (
+        f"ssh -nNT -L {local_port}:{compute_node}:{remote_port} {username}@{gateway}"
```
###  how about using ControlMaster and ControlPath while creating the connection
"ssh -nNT -M -S tmp_tunnel -L {local_port}:{compute_node}:{remote_port} {username}@{gateway}"

siligam replied:
I guess it will close the ports gracefully. I am not 100% sure but could give a try.

pgierz replied:
The ports close regularly after a Control + C, so I don't think we need to complicate the SSH command too much by using this.

siligam commented on src/pymorize/ssh_tunnel.py:
```diff
@@ -0,0 +1,40 @@
+    except subprocess.CalledProcessError as e:
+        click.echo(f"An error occurred while creating the SSH tunnel: {e}")
+    except KeyboardInterrupt:
+        click.echo("SSH tunnel closed.")
```
### now using the ControlPath to close the connection
subprocess.run(f"ssh -S ./tmp_tunnel -O exit {gateway}", shell=True, check=True)

pgierz replied:
Closing this comment, adding @siligam's suggestion above.

pgierz commented on src/pymorize/ssh_tunnel.py:
```diff
@@ -0,0 +1,40 @@
+    except subprocess.CalledProcessError as e:
+        click.echo(f"An error occurred while creating the SSH tunnel: {e}")
+    except KeyboardInterrupt:
+        click.echo("SSH tunnel closed.")
```
```suggestion
        subprocess.run(f"ssh -S ./tmp_tunnel -O exit {gateway}", shell=True, check=True)
        click.echo("SSH tunnel closed.")
```

pgierz commented:
This pull request updates the `ssh_tunnel.py` script to support forwarding ports for both Dask and Prefect dashboards. The changes include adding new options for specifying local and remote ports for Prefect, modifying the gateway server hostname, and updating the SSH command and output messages accordingly.

### Enhancements to SSH Tunnel Script:

* `src/pymorize/ssh_tunnel.py`: 
  - Added options for specifying local and remote ports for Prefect (`--local-prefect-port` and `--remote-prefect-port`).
  - Changed the default gateway server hostname from `levante2.dkrz.de` to `levante1.dkrz.de`.
  - Updated the SSH command to include port forwarding for Prefect in addition to Dask.
  - Modified the output messages to reflect the new port forwarding setup and provide URLs for accessing both Dask and Prefect dashboards.
